routers/repo/setting.go

```diff
@@ -727,3 +730,56 @@ func init() {
 		panic(err)
 	}
 }
+
+// UpdateAvatarSetting update repo's avatar
+func UpdateAvatarSetting(ctx *context.Context, form auth.AvatarForm) error {
+	ctxRepo := ctx.Repo.Repository
+	if form.Avatar != nil {
```

Comment:
Same here (do not use else):
```suggestion
	if form.Avatar == nil {
		// No avatar is uploaded but setting has been changed to enable
		// No random avatar here.
		if !com.IsFile(ctxRepo.CustomAvatarPath()) {
			log.Trace("No avatar was uploaded for repo: %d. Default icon will appear instead.", ctxRepo.ID)
		}
		return nil
	}
```

Reply:
Done in https://github.com/go-gitea/gitea/pull/6986/commits/f53758af4662b3d462677d92fb2b1ed472ce7998